editoast/src/views/rolling_stocks/mod.rs

```diff
@@ -405,33 +421,81 @@ async fn get_rolling_stock_usage(
     rolling_stock.get_rolling_stock_usage(conn).await
 }
 
+#[derive(ToSchema)]
+#[allow(unused)] // Schema only
+struct RollingStockLiveryCreateForm {
+    name: String,
+    images: Vec<Vec<u8>>,
+}
+
+async fn parse_mulipart_content(
+    mut form: Multipart,
+) -> Result<(String, Vec<MultipartImage>), LiveryMultipartError> {
+    let mut name = None;
+    let mut images = Vec::new();
+    while let Some(field) = form.next_field().await? {
+        let field_name = field.name().ok_or(LiveryMultipartError::MissingFieldName)?;
+        if field_name == "name" {
+            name = Some(
+                field
+                    .text()
+                    .await
+                    .map_err(|source| LiveryMultipartError::InvalidName { source })?,
+            );
+            continue;
+        }
+        if field_name != "images" {
+            return Err(LiveryMultipartError::UnrecognizedField {
+                field_name: field_name.to_owned(),
+            });
+        }
+        let file_name = field
+            .file_name()
+            .ok_or(LiveryMultipartError::MissingLiveryName)?
+            .to_owned();
+        let data = field
+            .bytes()
+            .await
+            .map_err(|source| LiveryMultipartError::InvalidLiveryContent { source })?
+            .to_vec();
+        images.push(MultipartImage { file_name, data });
+    }
+    if images.is_empty() {
+        return Err(LiveryMultipartError::MissingLiveryContent);
+    }
+    Ok((name.ok_or(LiveryMultipartError::MissingFieldName)?, images))
+}
+
 /// Create a rolling stock livery
-#[utoipa::path(tag = "rolling_stock,rolling_stock_livery",
+#[utoipa::path(
+    post, path = "",
+    tag = "rolling_stock,rolling_stock_livery",
     params(RollingStockIdParam),
     request_body = RollingStockLiveryCreateForm,
     responses(
         (status = 200, description = "The created rolling stock", body = RollingStockLivery),
         (status = 404, description = "The requested rolling stock was not found"),
     )
 )]
-#[post("")]
 async fn create_livery(
-    db_pool: Data<DbConnectionPoolV2>,
-    rolling_stock_id: Path<i64>,
-    MultipartForm(form): MultipartForm<RollingStockLiveryCreateForm>,
+    State(db_pool): State<DbConnectionPoolV2>,
+    Path(rolling_stock_id): Path<i64>,
+    form: Multipart,
 ) -> Result<Json<RollingStockLivery>> {
     let conn = &mut db_pool.get().await?;
 
-    let rolling_stock_id = rolling_stock_id.into_inner();
+    let (name, images) = parse_mulipart_content(form)
```

Comment:
I don't believe we did 😬

However, since we want to try to reduce the number of dependencies editoast has, I'm not sure if it's worth adding that for a single use case. (If one day we write another endpoint for a multipart request, I'm all for using the lib though.)

Wdyt?

Reply:
Yes, I almost didn't make the comment for that reason... ok fine, I'm resolving this comment.